libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -406,115 +161,501 @@ export class DynamicFormGroup<TModel> extends FormGroup {
+        }
+      });
+
+    // Nested Validation for the field for the requested class-validator group
+    const nestedGroupValidations: ValidationMetadata[] = [];
+    validationGroupMetadatas.forEach(validationMetadata => {
+      if (isPropertyValidatorOfType(validationMetadata, fieldName, ValidationKeys.nested.type)) {
+        nestedGroupValidations.push(validationMetadata);
+      }
+    });
+
+    const fieldDefinition: DynamicFormGroupField = {
+      data: formGroupFields[fieldName],
+      validation: []
+    };
+
+    if (fieldDefinition.data === undefined) {
+      fieldDefinition.data = fields[fieldName];
+    }
+
+    validationGroupMetadatas.forEach(validationMetadata => {
+      if (validationMetadata.propertyName === fieldName && validationMetadata.type !== ValidationKeys.conditional.type) {
+
+        // tslint:disable-next-line:forin
+        for (const typeKey in ValidationTypes) {
+
+          const canValidateField = ValidationTypes.hasOwnProperty(typeKey)
+            && validationMetadata.type === ValidationTypes[typeKey]
+            && validator[validationMetadata.type] === undefined;
+
+          if (canValidateField) {
+
+            const canValidateNested = (
+              (
+                allNestedValidations.length === nestedGroupValidations.length
+                || (fields[fieldName] instanceof DynamicFormGroup && allNestedValidations.length > 0 && nestedGroupValidations.length === 0)
+                || (fields[fieldName] instanceof FormArray && allNestedValidations.length > 0 && nestedGroupValidations.length === 0)
+              )
+              &&
+              (
+                // Property is marked with @ValidateNested()
+                validationMetadata.type === ValidationKeys.nested.type
+                && typeKey === ValidationKeys.nested.key
+              )
+            );
+
+            // Handle nested validation
+            if (canValidateNested) {
+              let objectToValidate: any;
+
+              if (fields[fieldName] instanceof DynamicFormGroup) {
+                fields[fieldName].object = fields[fieldName].fields;
+                objectToValidate = fields[fieldName].object;
+              }
+
+              // Fill field data if empty
+              if (fieldDefinition.data === undefined) {
+                fieldDefinition.data = fields[fieldName];
+              }
+
+              // create validation function and add to field validation
+              const nestedValidate = function (control: FormControl) {
+                const isValid = getValidateErrors(control, objectToValidate, validatorOptions).length === 0;
+                return getIsValidResult(isValid, validationMetadata);
+              };
+
+              fieldDefinition.validation.push(nestedValidate);
+            }
+
+            // Handle Custom Validation
+            else if (validationMetadata.type === ValidationKeys.custom.type && typeKey === ValidationKeys.custom.key) {
```

Comment:
Per validationMetadata, the type can only be one. 

Replaced successive `if()` with `else if()`